### DISCUSSION TITLE
Is the Nyquist term handle explicitly?

### DISCUSSION THREAD

**Discussion by Veenty** [JuliaMath/NFFT.jl#122] [Q&A]
I'm doing the map
$$\hat{f}_n  \to f(x_j)$$

using NFFT of the first kind (using $-k$ in the argument) and I'm not really sure if the implementation considers the Nyquist term or should I handle it differently?

**Comment by JeffFessler** [comment]
What is the "Nyquist term"?  I recommend looking at the docs here:
https://juliamath.github.io/NFFT.jl/dev/background/
Then if you are still unsure, try asking the question in a way that is related to the documentation.
If needed, we can update the documentation to clarify.

**Reply by Veenty** [reply]
When you are taking a discrete Fourier transform (real space to frequency space), so that the DFT generates

$$ \left( f(x_i) \right) \to (\hat{f}_n)_{n} $$
i.e. $hat{f}_n$ are the approximations of the Fourier coefficients of $f$, then if the number of samples of $f$ (denoted by $N$), is even, the best trigonometric approximation of $f$ is given by

$$f \approx \sum_{n= -\frac{N-2}{2} }^{\frac{N-2}{2}}  \hat{f}_n e^{2\pi i x n} + \hat{f}_{N/2} \cos(2\pi x \frac{N}{2})   $$

That last term with the cosine is the Nyquist term. So generally if you want to take derivatives or do 0 padding, that term needs to be handled in a special way. 


Edit: I had some hats missing

**Reply by JeffFessler** [reply]
I am not familiar with that expression being the best approximation, though does not seem to render properly in my browser so I may be misinterpreting it.  AFAIK, this package does not do anything special with the last term.  And this package is used for MRI where the signals are complex valued and I wonder if that cos expression is optimal for complex signals.

**Reply by Veenty** [reply]
Sorry don't know what happened with the rendering.

Here is are some notes by the guy that wrote FFTW https://math.mit.edu/~stevenj/fft-deriv.pdf 
(Page 4 right before section 3)

Basically the result of the Nyquist term is true for real and complex signals.

**Comment by Veenty** [comment]
The expression is:


$$f(x) \approx \sum_{n= -\frac{N-2}{2} }^{\frac{N-2}{2}} \hat{f}[n] e^{2\pi i x n} + \hat{f}[N/2]  \cos(2\pi x \frac{N}{2}) $$


It renders outside of the replays?
